### DISCUSSION TITLE
Bug in material points specification during oedometer 3D example

### DISCUSSION THREAD

**Discussion by brunohoo** [Anura3D/Anura3D_OpenSource#223] [General]
Hi all,

I am new to MPM and Anura3D, but knowledgeable on mesh-based numerical methods. I've been trying to do the first exercise in the tutorial manual. It seems that there is a bug when Anura3D assigns material points to the respective tetrahedron element during the GOD file creation.

My material point specification is correctly defined in GiD for the lower column:

![Printscreen1](https://github.com/Anura3D/Anura3D_OpenSource/assets/68155911/b4b3dce8-afa8-47e3-ae2f-a6a2b70d2642)

And my mesh was correctly generated with 66 tetrahedron elements and 207 nodes.

However, post-processing reveals that the material points are wrongly placed:

![Printscreen2](https://github.com/Anura3D/Anura3D_OpenSource/assets/68155911/6cfca995-3027-49e4-a387-18432d12aa97)

Thus making the simulation behave unexpectedly.

I checked the GOD file and it indeed seems to be a problem with the association of tetrahedron elements with the material point specification.

For example:

* My first line of `$$STARTELEMCON` is `31 11 22 36 18 14 27 30 24 28`.
* The first line of `$$STARTELMMAT` is `0`.
* The corresponding first line of `$$START_NUMBER_OF_MATERIAL_POINTS` is `0 0`.

When highlighting the respective Point IDs (subtracting one, because IDs in Paraview seem to start with zero, while node coordinates in GiD/Anura3D start with 1), it indeed seems that the plane that is associated with this first line of element connectivities is one of the elements lacking material points:

![Printscreen3](https://github.com/Anura3D/Anura3D_OpenSource/assets/68155911/2837e9d2-18e4-4f90-b591-b29c8fdaabe5)

I am using GiD x64 16.0.2 and the Anura3D 2023 version. The oedometer 2D example ran just fine. The GiD file associated with this faulty 3D example is here:

[oedometer.gid.zip](https://github.com/Anura3D/Anura3D_OpenSource/files/15310297/oedometer.gid.zip)

I also made a video recording my modeling process and the error being generated: https://youtu.be/JJvY3hAP460

Would gladly appreciate any help. Super excited to learn more. Thanks!

**Comment by FrancescaCeccato** [comment]
It seems that GiD is not assigning correctly the material and material points to elements. Did you try to re-generate the mesh and check that you do not have other elements than tetrahedra (tiangles or lines)? After re-generating the mesh try to regenerate Anura input.
